src/cli/telemetry/telemetry.ts

```diff
@@ -170,20 +175,67 @@ async function getInstalledPackages(sys: d.CompilerSystem, config: d.Config): Pr
       ([k]) => k.startsWith('@stencil/') || k.startsWith('@ionic/') || k.startsWith('@capacitor/')
     );
 
-    packages = packageLockJson
-      ? ionicPackages.map(
-          ([k, v]) =>
-            `${k}@${packageLockJson?.dependencies[k]?.version ?? packageLockJson?.devDependencies[k]?.version ?? v}`
-        )
-      : ionicPackages.map(([k, v]) => `${k}@${v}`);
+    try {
+      if (!isUsingYarn(sys)) {
+        packages = await npmPackages(sys, ionicPackages);
+      } else if (isUsingYarn(sys)) {
+        packages = await yarnPackages(sys, ionicPackages);
+      } else {
+        packages = ionicPackages.map(([k, v]) => `${k}@${v.replace('^', '')}`);
+      }
+    } catch (e) {
+      packages = ionicPackages.map(([k, v]) => `${k}@${v.replace('^', '')}`);
+    }
+
+    packages_no_versions = ionicPackages.map(([k]) => `${k}`);
 
-    return { packages };
+    return { packages, packages_no_versions };
   } catch (err) {
     hasDebug(config) && console.error(err);
-    return { packages };
+    return { packages, packages_no_versions };
   }
 }
 
+/**
+ * Visits the npm lock file to find the exact versions that are installed
+ * @param sys The system where the command is invoked
+ * @param ionicPackages a list of the found packages matching `@stencil`, `@capacitor`, or `@ionic` from the package.json file.
+ * @returns an array of strings of all the packages and their versions.
+ */
+async function npmPackages(sys: d.CompilerSystem, ionicPackages: [string, string][]) {
+  const appRootDir = sys.getCurrentDirectory();
+  const packageLockJson: any = await tryFn(readJson, sys, sys.resolvePath(appRootDir + '/package-lock.json'));
```

Comment:
We should use `path.join` (or the sys equivalent) here, to make the path delimiter OS-agonstic